vault/resource_kmip_secret_backend.go

```diff
@@ -133,17 +134,39 @@ func kmipSecretBackendUpdate(d *schema.ResourceData, meta interface{}) error {
 	path := d.Id()
 
 	if !d.IsNewResource() && d.HasChange("path") {
-		newPath := d.Get("path").(string)
+		src := path
+		dest := d.Get("path").(string)
 
-		log.Printf("[DEBUG] Remount %s to %s in Vault", path, newPath)
+		log.Printf("[DEBUG] Remount %s to %s in Vault", src, dest)
 
-		err := client.Sys().Remount(path, newPath)
+		err := client.Sys().Remount(src, dest)
 		if err != nil {
 			return fmt.Errorf("error remounting in Vault: %s", err)
 		}
 
-		d.SetId(newPath)
-		path = newPath
+		// There is something similar in resource_mount.go, but in the call to TuneMount().
+		var tries int
+		for {
```

Comment:
Is this  mount check loop something we can/should refactor to a function? I think this might be a useful check to add to other similar resources that also setup their own mounts; since we plan to support remounts for resources such as `vault_aws_secret_backend` in the future via adaptive migration

Reply:
Nice, was hoping you would catch that! Yeah, for sure. I thought you might do that in your secrets remount work.

Reply:
Nice! Sounds good, I will make a note to make this a part of that work :)